letsencrypt/plugins/common.py

```diff
@@ -136,7 +155,41 @@ def get_port(self):
 
     def get_addr_obj(self, port):
         """Return new address object with same addr and new port."""
-        return self.__class__((self.tup[0], port))
+        return self.__class__((self.tup[0], port), self.ipv6)
+
+    def _normalize_ipv6(self, addr):
+        """Return IPv6 address in normalized form, helper function"""
+        addr = addr.lstrip("[")
+        addr = addr.rstrip("]")
+        return self._explode_ipv6(addr)
+
+    def get_ipv6_exploded(self):
+        """Return IPv6 in normalized form"""
+        if self.ipv6:
+            return ":".join(self._normalize_ipv6(self.tup[0]))
+        return ""
+
+    def _explode_ipv6(self, addr):
+        """Explode IPv6 address for comparison"""
+        result = ['0', '0', '0', '0', '0', '0', '0', '0']
```

Comment:
nit: it might be easier to read this as `result = ["0"]*8` though I'm not sure how python 3 will like this